### PR TITLE
Dashboard SQL - General Overview data #121

### LINKED ISSUE #121: Dashboard SQL - General Overview data
### Dashboard SQL - General Overview data

Base: `main`
New branch: `sql-121-general-overview`

---

**Prerequisites**:

- [x] Finish reading the entire issue before beninging.

---

**Requirements**:

- In `src\repositories\sql\_dashboard` (from this point on will be referred to as **Dashboard SQL folder)**:
  - In the file `general_overview.sql`, add a SELECT command that will return these fields:
    - `reported_users` - Number of reported users
    - `reported_posts` - Number of reported posts
    - `reported_comments` - Number of reported comments
    - `pending_posts` - Number of pending posts
    - `banned_users` - Number of banned users
    - `disabled_posts` - Number of disabled posts
    - `disabled_comments` - Number of disabled comments
    - `issued_at` - Current date, aka `NOW()`

---

**Notes**:

- This SQL command will be used for this screen
image

### PR DESCRIPTION
**Pull request by LyndonTomas**
Closes #121

Tables Affected in the query:
tables affected and their contents

Query Sample:
query sample

If there are any questions or revisions that you'd like me to do, let me know.

### CODE REVIEW

**Review by jp-ong — CHANGES_REQUESTED** [review]
Nice nice.

I guess since we're adding a new configuration (create view) to the database...
- Add the create view command to `create.sql`
- Add the drop view command to `drop.sql`

**Comment by LyndonTomas** [comment]
> Nice nice.
> 
> I guess since we're adding a new configuration (create view) to the database...
> 
> * Add the create view command to `create.sql`
> * Add the drop view command to `drop.sql`

Sure, just keep in mind that we need to run drop and create function every time you need to "refresh".

**Comment by jp-ong** [comment]
> > Nice nice.
> > I guess since we're adding a new configuration (create view) to the database...
> > 
> > * Add the create view command to `create.sql`
> > * Add the drop view command to `drop.sql`
> 
> Sure, just keep in mind that we need to run drop and create function every time you need to "refresh".

Why? Doesn't seem like we need to. Image below shows the view being created once then being executed multiple times after. Creating a view is the same as creating a table or a rule.
image

File: src/repositories/sql/create.sql [hunk 1]
```diff
@@ -194,4 +194,18 @@ CREATE OR REPLACE RULE count_comments AS
...
+    SELECT 
+        (SELECT COUNT (DISTINCT target_user_id) FROM reports) AS reported_users,
+        (SELECT COUNT (DISTINCT target_post_id) FROM reports) AS reported_posts,
+        (SELECT COUNT (DISTINCT target_comment_id) FROM reports) AS reported_comments,
+        (SELECT COUNT (target_id) FROM bans WHERE bans.expires_at > now()) AS banned_users,
+        (SELECT COUNT (post_id) FROM posts WHERE posts.status_id = 'ps1') AS pending_posts,
+        (SELECT COUNT (post_id) FROM posts WHERE posts.status_id = 'ps4') AS disabled_posts,
+        (SELECT COUNT (comment_id) FROM comments WHERE is_disabled = TRUE) AS disabled_comments,
+        (SELECT now()) AS issued_at;
+SELECT * FROM general_overview;
```
Comment (jp-ong) [inline src/repositories/sql/create.sql:211]: I don't think this belongs here.

**Review by jp-ong — CHANGES_REQUESTED** [review]

File: src/repositories/sql/create.sql (hunk 1 above)
Comment (LyndonTomas) [reply src/repositories/sql/create.sql:211]: Yes, thank you for reminding me. I accidentally added it there so the output will be seen.

**Review by jp-ong — APPROVED** [review]

**Pull request merged**